### PR TITLE
Describe forms creation and customization

### LINKED ISSUE #944: [feature] Allow using multiple columns in attribute forms
Original commit: https://github.com/qgis/QGIS/commit/faf6b2654cf4887e093f4beb4cd6121fa78b4151 by m-kuhn

When using the drag and drop designer, a user can specify over how many
columns the fields should be distributed.

A double click on an existing group will allow adapting the value.

### LINKED ISSUE #1313: [feature] Allow controlling labels for individual edit widgets
Original commit: https://github.com/qgis/QGIS/commit/f10f4a5a12bd2bc23cd8f98eacdf42ab5caca18d by m-kuhn

In the drag and drop designer, a double click on an item will allow
controlling if the label should be shown for each item individually.

Fix #15450

### LINKED ISSUE #1399: [Feature] conditional visibility for tabs and groupboxes
Original commit: https://github.com/qgis/QGIS/commit/46606b2d775df0f56e302283ff790d426f775c9f by m-kuhn

This adds a new configuration option to conditionally show or hide
tabs and groupboxes in drag and drop designer forms.

Configuration is done via a double click in the designer tree in the
fields configuration interface.

An expression can be entered to control the visibility. The expression
will be re-evaluated everytime values in the form change and the tab or
groupbox shown/hidden accordingly.

### LINKED ISSUE #1415: [Feature] Allow configuring link/unlink feature buttons on relation editor widget
Original commit: https://github.com/qgis/QGIS/commit/5f0ade83f3ae989abb607f7f185ca0849b5b11dd by m-kuhn

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by DelazJ**
This PR structures and describes features inside Layer properties --> Fields tab (form customization, edit widget). 
fix #1399, fix #944, fix #1313, fix #1415 
ref #705, #1365 

Review welcome...

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -1409,11 +1424,39 @@ the alias.
...
+.. index:: Default values, Fields constraints
+
+Common settings
+...............
+
+Regardless the type of widget applied to the field, there are some common
+properties you can set:
+
+* **Editable**: uncheck this to set the field read-only (not manually
+  modifiable) when the layer is in edit mode. Note that setting the field
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: by 'the field editable' you mean 'the field named editable'? Rahter a french translation sentence, isn't? May be you can make the sentence shorter with `Note that this setting doesn't override a ...`

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -1409,11 +1424,39 @@ the alias.
...
+  editable doesn't override any edit limitation from the provider.
+* **Label on top**: places the field name above or beside the widget in the
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Proposition: `places the field name above the widget in the feature form, default is beside the widget.`

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -1409,11 +1424,39 @@ the alias.
...
+  feature form
+* **Default value**: for new features, populates by default the field with the
+  same value or an expression-based one. For example, you can use
+  ``maximum("field")+1`` to generate a field that increments by 1 for each
+  new feature. A preview of the default value is displayed
+* **Constraints**: you can constrain the value filled in the field.
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: `filled`:  is it not so obvious?

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -1409,11 +1424,39 @@ the alias.
...
+  First, you can enable the |checkbox| :guilabel:`Not null` checkbox to
+  force the user to fill the value of the column.
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: s/column/field as we are in a form, isn't?

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -1409,11 +1424,39 @@ the alias.
...
+
+  You can also setup a custom contraint using an expression, e.g.
+  ``regexp_match(col0,'A-Za-z')`` to be sure that the value of the column *col0*
+  has only alphabetical letter. Finally, you can add a short description to explain
+  the constraint that will be displayed in the top of the form as a warning
+  message when the value of the field does not match the constraint.
+
+
+Edit widget
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: widgetS?

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 6]
```diff
@@ -1485,51 +1518,48 @@ message when the value of the field does not match the constraint.
...
+the result of the default ``Autogenerate`` value of the :menuselection:`Layer
+properties --> Fields --> Attribute editor layout` setting. Thanks to the
+:ref:`widget setting <configure_field>`, you can improve this dialog.
 
-* The 'Provide ui-file' option allows you to use complex dialogs made with the Qt-Designer.
-  Using a UI-file allows a great deal of freedom in creating a dialog.
-  For detailed information, see
-  http://nathanw.net/2011/09/05/qgis-tips-custom-feature-forms-with-python-logic/.
+You can furthermore define built-in forms (see figure_fields_form_).
+This is useful for data entry jobs or to identify features using the option
```
Comment (yjacolin) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: I don't understand what is data entry jobs here!

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 1 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: > by 'the field editable' you mean 'the field named editable'?

no, I meant checking the `Editable` option of the field widget.
What about "Note that checking this setting doesn't override any edit limitation from the provider."? I think that  it should be clear that, more than the setting itself it's its checked state that shouldn't be considered as a replacement of provider limitation.

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 2 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: not sure I get the "plus-value" of the change but, if you want...

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 3 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: 😆 will remove

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 5 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: ok

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 6 above)
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: I picked it from the previous formulation (line 1489). I guess it's about work sessions when one needs to fill many times attributes of features (e.g., offline edit...)

Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: I removed this sentence as it is not specific to built-in forms imho.

**Comment by yjacolin** [comment]
Thanks @DelazJ !

**Comment by DelazJ** [comment]
Thanks for the review!!

**Pull request merged**